python/ray/experimental/placement_group.py

```diff
@@ -1,15 +1,73 @@
+import random
 from typing import (List, Dict)
 
 import ray
+from ray._raylet import PlacementGroupID
 
 
 class PlacementGroup:
     """A handle to a placement group.
+
+    Args:
+        id: Placement group id.
+        bundles: List of bundles.
     """
 
-    def __init__(self, id, bundle_count):
+    @staticmethod
+    def empty():
+        return PlacementGroup(PlacementGroupID.nil(), [])
+
+    def __init__(self, id: PlacementGroupID, bundles: List[Dict[str, float]]):
         self.id = id
-        self.bundle_count = bundle_count
+        self.bundles = bundles
+
+    def ready(self):
+        """Returns an object ID to check ready status."""
+        worker = ray.worker.global_worker
+        worker.check_connected()
+
+        @ray.remote(num_cpus=0, max_calls=0)
+        def bundle_reservation_check(placement_group):
+            return placement_group
+
+        assert len(self.bundles) != 0, (
+            "ready() cannot be called on placement group object with a "
+            f"bundle length == 0, current bundle length: {len(self.bundles)}")
+
+        # Select a random bundle to schedule a dummy task.
+        # Since the placement group creation will be atomic, it is sufficient
+        # to schedule a single task.
+        bundle_index = random.randint(0, len(self.bundles) - 1)
+        bundle = self.bundles[bundle_index]
+
+        resource_name, value = self._get_none_zero_resource(bundle)
+        num_cpus = 0
+        num_gpus = 0
+        resources = None
+        if resource_name == "CPU":
+            num_cpus = value
+        elif resource_name == "GPU":
+            num_gpus = value
```

Comment:
Can't we always use resources instead of num_cpus etc?

Reply:
That was actually my original approach, but it is disallowed (https://github.com/ray-project/ray/blob/a82124d304a28c06732765c3a5806a817bed2552/python/ray/parameter.py#L278). I can also remove this check, but since we only have 2 of them, I decided to go to this approach. Please comment if you'd prefer to remove those error check.